src/components/DevfileEditor/__tests__/DevfileEditor.spec.tsx

```diff
@@ -17,10 +17,53 @@ import DevfileEditor from '../';
 import { BrandingData } from '../../../services/bootstrap/branding.constant';
 import { FakeStoreBuilder } from '../../../store/__mocks__/storeBuilder';
 import { createFakeCheWorkspace } from '../../../store/__mocks__/workspace';
-import { languages, editor } from 'monaco-editor-core/esm/vs/editor/editor.main';
 
-jest.mock('../../../../node_modules/monaco-editor-core/esm/vs/editor/editor.main', () => {
-  return () => ({ languages, editor });
+jest.mock('monaco-editor-core/esm/vs/editor/editor.main', () => {
+  return {
+    languages: {
+      registerCompletionItemProvider: jest.fn(),
+      registerDocumentSymbolProvider: jest.fn(),
+      registerHoverProvider: jest.fn(),
+      register: jest.fn(),
+      setMonarchTokensProvider: jest.fn(),
+      setLanguageConfiguration: jest.fn(),
+      CompletionItemInsertTextRule: {
+        InsertAsSnippet: jest.fn()
+      }
+    },
+    editor: {
+      create: jest.fn(),
+      getModel: jest.fn(),
+      getValue: jest.fn(),
+      setModelMarkers: jest.fn(),
+      defineTheme: jest.fn(),
+      setTheme: jest.fn()
+    }
+  };
+});
+
+jest.mock('../../../services/monacoThemeRegister', () => {
+  return {
+    initDefaultEditorTheme: () => ''
+  };
+});
+
+jest.mock('../monaco-converter', () => {
+  class ProtocolToMonacoConverter {
+    asCompletionResult = jest.fn();
+    asCompletionItem = jest.fn();
+    asSymbolInformations = jest.fn();
+    asHover = jest.fn();
+    asDiagnostics = jest.fn();
+  }
+  class MonacoToProtocolConverter {
+    asPosition = jest.fn();
+    asCompletionItem = jest.fn();
+  }
+  return {
+    ProtocolToMonacoConverter,
+    MonacoToProtocolConverter
+  };
```

Comment:
These changes seem redundant to me, the test only has to be slightly updated to pass. See https://github.com/eclipse-che/che-dashboard/blob/yarn2-migration/src/components/DevfileEditor/__tests__/DevfileEditor.spec.tsx#L22-L24

Reply:
Existing implementation 

```
import { languages, editor } from 'monaco-editor-core/esm/vs/editor/editor.main';

jest.mock('../../../../node_modules/monaco-editor-core/esm/vs/editor/editor.main', () => {
  return () => ({ languages, editor });
});
```
was done in a wrong way. I just fixed it.

Do you have another opinion about it?

Reply:
It will be really nice if you provide more details about what was wrong.

Reply:
maybe, it would be even better to move these changes into a separate PR for a more convenient discussion.

Reply:
I just provide existing mock in a right way because the problem has been identified after migrating to yarn 2.
Let discuss it here(this jest.mock for 'monaco-editor-core/esm/vs/editor/editor.main').

Reply:
The problem: 
```
import { languages, editor } from 'monaco-editor-core/esm/vs/editor/editor.main';

jest.mock('../../../../node_modules/monaco-editor-core/esm/vs/editor/editor.main', () => {
  return () => ({ languages, editor });
});
```
existing mock for  tried to get values from themselves.

It was a bug.

Reply:
Now I got it, thank you for the explanation.